packages/cli/src/commands/hydrogen/link.ts

```diff
@@ -0,0 +1,126 @@
+import Command from '@shopify/cli-kit/node/base-command';
+import {
+  renderConfirmationPrompt,
+  renderSelectPrompt,
+  renderWarning,
+} from '@shopify/cli-kit/node/ui';
+import {outputInfo, outputSuccess} from '@shopify/cli-kit/node/output';
+
+import {adminRequest, parseGid} from '../../lib/graphql.js';
+import {commonFlags} from '../../lib/flags.js';
+import {getHydrogenShop} from '../../lib/shop.js';
+import {getAdminSession} from '../../lib/admin-session.js';
+import {hydrogenStorefrontUrl} from '../../lib/admin-urls.js';
+import {
+  LinkStorefrontQuery,
+  LinkStorefrontSchema,
+} from '../../lib/graphql/admin/link-storefront.js';
+import {getConfig, setStorefront} from '../../lib/shopify-config.js';
+import {missingStorefronts} from '../../lib/missing-storefronts.js';
+
+export default class Link extends Command {
+  static description =
+    "Link your local development to one of your shop's Hydrogen storefronts.";
+
+  static hidden = true;
+
+  static flags = {
+    force: commonFlags.force,
+    path: commonFlags.path,
+    shop: commonFlags.shop,
+    storefront: commonFlags.hydrogenStorefront,
+  };
+
+  async run(): Promise<void> {
+    const {flags} = await this.parse(Link);
+    await linkStorefront(flags);
+  }
+}
+
+export interface LinkFlags {
+  force?: boolean;
+  path?: string;
+  shop?: string;
+  storefront?: string;
+}
+
+export async function linkStorefront({
+  force,
+  path,
+  shop: flagShop,
+  storefront: flagStorefront,
+}: LinkFlags) {
+  const shop = await getHydrogenShop({path, shop: flagShop});
+  const {storefront: configStorefront} = await getConfig(path ?? process.cwd());
+
+  if (configStorefront && !force) {
+    const overwriteLink = await renderConfirmationPrompt({
+      message: `${configStorefront.title} is currently linked to your local environment. Do you want to change storefronts?`,
```

Comment:
Maybe `local project` here too?

Wondering if we need to introduce the idea and terminology of local/remote, or if we can get away with `storefront` as the stand-in for the idea of "remote"

```suggestion
      message: `Your project is currently linked to ${configStorefront.title}. Do you want to re-link to a different Hydrogen storefront?`,
```

Reply:
Yes, I've been struggling with the terminology here because I don't want to keep saying "Hydrogen storefront defined in the admin". Maybe we can include the word `Shopify` to imply that it's a record we hold?

> Your project is currently linked to ${title}. Do you want to re-link to a different Hydrogen storefront on Shopify?

Does "on Shopify" make sense?